### PR TITLE
Apa pages

### PR DESCRIPTION
**Pull request by vuyaniShabangu**
Current work on APA pages designed using Ant as per the Hominum designs.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -32,6 +33,7 @@ module.exports = {
 			},
 		],
 		'jsx-a11y/href-no-hash': 'off',
+		'import/no-unresolved': 'off',
```
Comment (pvanheus) [inline .eslintrc.js]: This is not needed if you fix the import problems in pages/apa (index.tsx, pathogens/[pathogen-id].tsx and pathogens/index.tsx)

File: .eslintrc.js [hunk 2]
```diff
@@ -56,6 +59,10 @@ module.exports = {
 		],
 		'@typescript-eslint/no-explicit-any': 'off',
 		'@typescript-eslint/no-var-requires': 'off',
+		'@typescript-eslint/ban-types': 'off',
```
Comment (pvanheus) [inline .eslintrc.js:61]: this is not needed.

File: components/Button.tsx [hunk 3]
```diff
@@ -27,123 +27,123 @@ import defaultTheme from './theme';
 import { Spinner } from './theme/icons';
 
 export const UnStyledButton = styled('button')`
-  background: transparent;
```
Comment (pvanheus) [inline components/Button.tsx:30]: We need to decide on tabs vs spaces and indent standard. The changes here are simply whitespace.

File: pages/apa/pathogens/[pathogen-id].tsx
Comment (pvanheus) [inline pages/apa/pathogens/[pathogen-id].tsx:1]: Why the name with [ ] in it?

File: pages/apa/pathogens/[pathogen-id].tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/*import App from '../../components/pages/apa';
+
+export default App;*/
+
+import { createPage } from 'global/utils/pages';
```
Comment (pvanheus) [inline pages/apa/pathogens/[pathogen-id].tsx]: import '@global/utils/pages'; or '../../../global/utils/pages';

File: pages/apa/pathogens/index.tsx [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { createPage } from 'global/utils/pages';
```
Comment (pvanheus) [inline pages/apa/pathogens/index.tsx]: import '@global/utils/pages'; or '../../../global/utils/pages';

File: pages/apa/index.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
+/*import App from '../../components/pages/apa';
+
+export default App;*/
+
+import { createPage } from 'global/utils/pages';
```
Comment (pvanheus) [inline pages/apa/index.tsx]: import '../../global/utils/pages' or '@global/utils/pages'

File: .eslintrc.js (hunk 2 above)
Comment (vuyaniShabangu) [reply .eslintrc.js:61]: I get quite a few errors when this is taken out:

`Users/vuyani/Documents/sanbi/portal/global/hooks/useMuseData/index.ts
  74:15  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types

/Users/vuyani/Documents/sanbi/portal/global/utils/pages/index.tsx
  30:13  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/vuyani/Documents/sanbi/portal/global/utils/processStream.ts
  28:21  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types

✖ 3 problems (3 errors, 0 warnings)

vuyani@Vuyanis-MacBook-Pro portal % npm run lint

> virusseq-dataportal-ui@1.14.0 lint
> eslint --ext .ts,.tsx '.'

vuyani@Vuyanis-MacBook-Pro portal % npm run lint

> virusseq-dataportal-ui@1.14.0 lint
> eslint --ext .ts,.tsx '.'


/Users/vuyani/Documents/sanbi/portal/global/hooks/useMuseData/index.ts
  74:15  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types

/Users/vuyani/Documents/sanbi/portal/global/utils/pages/index.tsx
  30:13  error  Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- If you want a type meaning "empty object", you probably want `Record<string, never>` instead  @typescript-eslint/ban-types

/Users/vuyani/Documents/sanbi/portal/global/utils/processStream.ts
  28:21  error  Don't use `Function` as a type. The `Function` type accepts any function-like value.
It provides no type safety when calling the function, which can be a common source of bugs.
It also accepts things like class declarations, which will throw at runtime as they will not be called with `new`.
If you are expecting the function to accept certain arguments, you should explicitly define the function shape  @typescript-eslint/ban-types

✖ 3 problems (3 errors, 0 warnings)`

File: components/Button.tsx (hunk 3 above)
Comment (vuyaniShabangu) [reply components/Button.tsx:30]: True. But I think this was changed when I ran the linter and instructed it to fix the errors it finds.
I believe it takes this instruction from the `useTabs: true` rule in .eslintrc.js file.
Changing useTabs to false results in about 11529 problems when the linter runs. Most of these are warnings but cause a lot of noise in our builds. 

I think the project is meant to use tabs, it just some spaces made it in somehow, and this PR fixes those few instances.

File: pages/apa/pathogens/[pathogen-id].tsx
Comment (vuyaniShabangu) [reply pages/apa/pathogens/[pathogen-id].tsx:1]: This is actually a pretty cool feature of NextJS called Dynamic Routing.
More about this here: https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes

But basically, we are defining the page that will be hit when the localhost:/3000/pathogens/{some-pathogen-id-here} is hit, then in that page component, we receive the pathogen id as a prop, and can go fetch info about that pathogen in our backend. All while the page renders :)

File: pages/apa/index.tsx (hunk 6 above)
Comment (vuyaniShabangu) [reply pages/apa/index.tsx]: Resolved

File: pages/apa/pathogens/index.tsx (hunk 5 above)
Comment (vuyaniShabangu) [reply pages/apa/pathogens/index.tsx]: Resolved

File: pages/apa/pathogens/[pathogen-id].tsx (hunk 4 above)
Comment (vuyaniShabangu) [reply pages/apa/pathogens/[pathogen-id].tsx]: Resolved

File: .eslintrc.js (hunk 1 above)
Comment (vuyaniShabangu) [reply .eslintrc.js]: Resolved

File: .eslintrc.js (hunk 2 above)
Comment (pvanheus) [reply .eslintrc.js:61]: Ok odd.. I didn't get these errors. Well leave it in for now. I presume that these errors are a result of using placeholder code.

File: components/Button.tsx (hunk 3 above)
Comment (pvanheus) [reply components/Button.tsx:30]: Let's leave it with what makes the linter happy. :)

File: pages/apa/pathogens/[pathogen-id].tsx
Comment (pvanheus) [reply pages/apa/pathogens/[pathogen-id].tsx:1]: That is cool. Pity it kind of breaks Linux filenames.

**Pull request merged**